sql-constants.md

```diff
@@ -0,0 +1,220 @@
+---
+title: SQL Constant values
+summary: SQL Constants represent a simple value that doesn't change.
+toc: false
+---
+
+SQL Constants represent a simple value that doesn't change.
+
+<div id="toc"></div>
+
+## Introduction
+
+There are five categories of constants in CockroachDB:
+
+- string literals: these define string values but their actual data type will
+  be inferred from context, for example `'hello'`;
+- numeric literals: these define numeric values but their actual data
+  type will be inferred from context, for example `-12.3`;
+- byte array literals: these define byte array values with data type
+  `BYTES`, for example `b'hello'`;
+- interpreted literals: these define arbitrary values with an explicit
+  type, for example `INTERVAL '3 days'`.
+- named constants: these have predefined values with a predefined
+  type, for example `TRUE` or `NULL`.
+
+## String literals
+
+CockroachDB supports multiple formats for string literals:
+
+- standard SQL string literals
+- string literals with C escape sequences
+- hexadecimal-encoded strings
+
+The first two formats also allow arbitrary Unicode characters encoded as UTF-8.
+
+In all these cases, the actual data type of a string literal is determined 
+using the context where it appears.
+
+For example:
+
+| Expression | Data type of the string literal |
+|------------|---------------------------------|
+| `length('hello')` | `STRING` |
+| `now() + '3 day'`  | `INTERVAL` |
+| `INSERT INTO tb(date_col) VALUES ('2013-01-02')` | `DATE` |
+
+In general, the data type of a string literal is that demanded by the
+context if there is no ambiguity, or `STRING` otherwise.
+
+Check our blog for
+[more information about the typing of string literals](https://www.cockroachlabs.com/blog/revisiting-sql-typing-in-cockroachdb/).
+
+### Standard SQL string literals
+
+SQL string literals are formed by an arbitrary sequence of characters
+enclosed between single quotes (`'`), for example `'hello world'`. 
+to include a single quote in the string, use a double single quote.
+
+For example:
+
+```sql
```

Comment:
Same comments as elsewhere:

- Use `~~~` for code blocks.
- Preface sql statements with `$` and end them with `;`.
- Put response in separate code block without syntax highlighting.